packages/material-ui-system/src/spacing.js

```diff
@@ -22,8 +22,28 @@ const directions = {
 // From 300,000 ops/sec
 // To 350,000 ops/sec
 const getCssProperties = memoize(prop => {
+  switch (prop) {
+    case 'marginX':
+      prop = 'mx';
+      break;
+
+    case 'marginY':
+      prop = 'mx';
+      break;
+
+    case 'paddingX':
+      prop = 'px';
+      break;
+
+    case 'paddingY':
+      prop = 'py';
+      break;
+
+    default:
+  }
+
   // It's not a shorthand notation.
-  if (prop.length > 3) {
+  if (prop.length > 2) {
```

Comment:
We can be smarter about it and move it in the length branch code.

Reply:
I'm not sure what you mean but, also... I was just changing this to be more "correct" since the original number (3) doesn't seem to apply anymore if it ever did and wasn't just a typo in the first place.